sphinx/texinputs/sphinx.sty

```diff
@@ -1164,26 +1164,19 @@
 % {fulllineitems} is the main environment for object descriptions.
 %
 \newcommand{\py@itemnewline}[1]{%
+  \kern\labelsep
   \@tempdima\linewidth
-  \advance\@tempdima \leftmargin\makebox[\@tempdima][l]{#1}%
+  \advance\@tempdima \labelwidth\makebox[\@tempdima][l]{#1}%
+  \kern-\labelsep
```

Comment:
the LaTeX kernel code will add `\labelsep` shift after the box, this compensates it.